### PR TITLE
PR Redo - Purge actions and filter

### PR DESCRIPTION
**Pull request by timotei-litespeed**
Add filters for post purge events
Add actions for each major purge event + parameters to some actions

### CODE REVIEW

File: src/purge.cls.php [hunk 1]
```diff
@@ -797,6 +810,7 @@ public function purge_url($url, $purge2 = false, $quite = false)
 		self::add($hash, $purge2);
 
 		!$quite && !defined('LITESPEED_PURGE_SILENT') && Admin_Display::success(sprintf(__('Purge url %s', 'litespeed-cache'), $val));
+		do_action('litespeed_purged_link');
```
Comment (hi-hai) [inline src/purge.cls.php:813]: @timotei-litespeed Maybe add the url as param?

Comment (timotei-litespeed) [reply src/purge.cls.php]: Added and looked over other functions and made a few changes

File: src/purge.cls.php [hunk 2]
```diff
@@ -924,11 +945,14 @@ public static function purge_comment_widget()
...
 			return;
 		}
+
 		$recent_comments = $wp_widget_factory->widgets['WP_Widget_Recent_Comments'];
 		if (!is_null($recent_comments)) {
 			self::add(Tag::TYPE_WIDGET . $recent_comments->id);
 			self::add_private(Tag::TYPE_WIDGET . $recent_comments->id);
 		}
+
+		do_action('litespeed_purged_comment_widget', $recent_comments->id);
```
Comment (hi-hai) [inline src/purge.cls.php:955]: This will result in error if you don't use the way L950 does.

File: src/purge.cls.php (hunk 1 above)
Comment (hi-hai) [reply src/purge.cls.php]: Good try. There are still some places you may miss. Also some comments added

**Pull request merged**